src/claimtrie.h

```diff
@@ -28,6 +34,34 @@
 
 uint256 getValueHash(COutPoint outPoint, int nHeightOfLastTakeover);
 
+// lower-case and normalize any input string name
+// see: https://unicode.org/reports/tr15/#Norm_Forms
+inline std::string normalizeClaimName(const std::string& name)
+{
+    static std::locale utf8;
+    static bool initialized = false;
+    if (!initialized) {
+        static boost::locale::localization_backend_manager manager =
+            boost::locale::localization_backend_manager::global();
+        manager.select("icu");
+
+        static boost::locale::generator curLocale(manager);
+        utf8 = curLocale("en_US.UTF8");
+        initialized = true;
+    }
+
+    std::string normalized = boost::locale::normalize(name, boost::locale::norm_nfd, utf8);
+    // Non-locale aware lowercase
+    boost::algorithm::to_lower(normalized);
+
+    // Locale aware lowercase
+    /* normalized = boost::locale::to_lower(normalized, utf8); */
+
+    LogPrintf("%s: Normalized \"%s\" (%lu, %lu) to \"%s\" (%lu, %lu)\n", __func__, name, name.size(), sizeof(name), normalized, normalized.size(), sizeof(normalized));
+    // code points can be lost if we're not careful to make sure the string isn't shrinking here
+    return (normalized.size() >= name.size()) ? normalized : name;
```

Comment:
I don't understand exactly what can cause a string to shrink here?

Reply:
That's for UTF-8 code points after normalization.  For example, 0xc0a0 and 0x20 are encodings of the same thing, but collapsing the former (due to the lower casing) is not handled as equivalent.  When there's a dispute, we use the original.  Perhaps we could opt to use something else in that case, but again, that's for consistency.

Reply:
> I don't understand exactly what can cause a string to shrink here? 

Also, technically it isn't a string as you normally think of them anymore.  It is a string of byte-length code points, which are variably sized to make the characters.  Sometimes a single encoded character may be N bytes, but after normalization it may be M.

Reply:
Let me rephrase, I get why the size of the un-normalized string could shrink after normalization, 

But I don't get why we are using that as the basis for determining whether we return the original un-normalized string, or the normalized string? Is there something special about this rule? Can you elaborate on your specific example of 0x20 and 0xc0a0 here? What are those characters and why aren't they handled as equivalent?